enhancements/20220530-vxlan-enhancement.md

```diff
@@ -0,0 +1,521 @@
+
+ ingress replication ( default value)
+ multicast replication (some physical network may not support multicast)
+ centralized replication
+
+#### Story 5
+
+VM migration
+
+![](./20220530-vxlan-enhancement/vxlan-vm-migration.png)
+
+When an VM is migrated/moved to another NODE, `Host Harvester Cluster` updates its info in VXLAN FDB and/or related Gateway/NAT/...
+
+VM does not need send gratuitous ARP of its (NODE) IP/MAC. VIP is discussed in [VIP](#vip-floating-in-vxlan)
+
+#### Story 6
+
+VM-VM communication in different Guest Kubernetes Cluster/ VM Group (different VNI)
+
+General speaking, different Guest Kubernetes Cluster / VM Group does not communicate with each directly. But that's still possible. In such case, a L3 VXLAN Gateway is needed.
+
+##### Centralized Gateway
+
+![](./20220530-vxlan-enhancement/vxlan-centralized-l3-gateway.png)
+
+Centralized VXLAN gateway deployment has the following advantages and disadvantages:
+
+Advantage:
+  
+  Inter-subnet traffic can be centrally managed.
+  
+  Gateway deployment and management are simple.
+
+Disadvantages:
+
+  Forwarding paths are not optimal. Inter-subnet Layer 3 traffic of the same Layer 2 gateway must be transmitted to the centralized Layer 3 gateway for forwarding.
+
+  The ARP entry may be too big, in worst cases, all VM's MAC are in there. (VIP can reduce the number)
+
+##### Distributed Gateway
+
+  A big topic, BGP EVPN and more are included.
+
+  TBD
+
+#### Story 7
+
+VM-internet communication
+
+For VM Group or Guest Kubernetes Cluster, they need to communicate with internet and vice versa.
+
+It is described together in story 9.
+
+#### Story 8
+
+Guest Kubernetes Cluster - Rancher Integration
+
+##### Guest Kubernetes Cluster creation
+
+https://docs.harvesterhci.io/v1.0/rancher/rke2-cluster/
+
+When creating an RKE2 Kubernetes Cluster via Harvester Node Driver, user select the reateld vxlan_id/vni_id . (vni may be created in advance or automatically)
+
+![](./20220530-vxlan-enhancement/guest-k8s-cluster-create-1.png)
+
+##### VM Management
+
+According to following document, the VM management is done in Harvester UI. Rancher document does not mention this part.
+
+https://docs.harvesterhci.io/v1.0/rancher/virtualization-management/
+
+Open Points:
+
+1. When a new VM is created for a specific Guest Kubernetes Cluster in Harvester, will it be added to this Cluster automatically ?
+
+2. When VM has no eth0 (management network) attached, can Harvester UI use WebRNC to connect it?
+
+##### Guest Kubernetes Cluster management
+
+Per Rancher document, Rancher will contact with Guest Kubernetes Cluster via CLI or RancherUI, it means, Rancher needs to be network reachable with Guest Kubernetes Cluster, at least it's VIP.
+
+https://rancher.com/docs/rancher/v2.6/en/cluster-admin/cluster-access/
+
+https://rancher.com/docs/rancher/v2.6/en/cluster-admin/cluster-access/kubectl/
+
+![](./20220530-vxlan-enhancement/rancher-integration-1.png)
+
+
+#### Story 9
```

Comment:
In story 9, you only describe ingress traffic. (From internet to VM)
How about egress traffic without EIP? (From VM to internet)
Will it have a router ip to perform NAPT (Network Address Port Transfer) or "masquerade NAT"?
It will be like your home router.
All your devices will be NATed as home router's ip and send to internet without any port forwarding setting.
And home router's IP is only one and shared with all devices.